homeassistant/components/legrand_rflc/light.py

```diff
@@ -0,0 +1,180 @@
+"""The Legrand RFLC integration light platform."""
+
+from collections.abc import Mapping
+from typing import Final
+
+from homeassistant import config_entries
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_TRANSITION,
+    COLOR_MODE_BRIGHTNESS,
+    SUPPORT_TRANSITION,
+    LightEntity,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+
+class _LegrandRflcSwitch(LightEntity):
+    _attr_should_poll = False
+
+    def __init__(self, hub, zid: int, properties: Mapping):
+        self._hub = hub
+        self._attr_name = properties[hub.NAME]
+        self._attr_is_on = properties[hub.POWER]
+        self._zid = zid
+        hub.on(
+            f"{hub.EVENT_ZONE_PROPERTIES_CHANGED}:{zid}",
+            self._zone_properties_changed,
+        )
+        hub.on(hub.EVENT_CONNECTED, self._available)
```

Comment:
State update callbacks must be registered after the entity has been added to Home Assistant in entity method `async_added_to_hass`. Please move these lines there.

Reply:
https://github.com/home-assistant/core/pull/53430/commits/9cc239b60b7bba88e5f9019a2bd5da37ce10b59e